chips/stm32f3xx/src/gpio.rs

```diff
@@ -472,6 +472,7 @@ impl PinId {
         unsafe { &PIN[usize::from(port_num)][usize::from(pin_num)] }
     }
 
+    #[allow(clippy::mut_from_ref)]
```

Comment:
Are we sure this is actually okay? If this was pointing directly to a hardware reference maybe that would be more clearly safe, but I think this code technically could allow multiple aliased references to the pin `Option`. I think this is actually identifying a real error here.

In general, I think we need comments on all the allows we add explaining why they're safe / allowed.

Reply:
I think you're right that this is a real error, and that we need comments on all of the allows for correctness-based lints